pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -160,15 +163,28 @@ __maybe_unused static __always_inline void protocol_classifier_entrypoint(struct
 
     protocol_t app_layer_proto = get_protocol_from_stack(protocol_stack, LAYER_APPLICATION);
 
-    if ((app_layer_proto == PROTOCOL_UNKNOWN || app_layer_proto == PROTOCOL_POSTGRES) && is_tls(buffer, usm_ctx->buffer.size, skb_info.data_end)) {
+    tls_record_header_t tls_hdr = {0};
+
+    if ((app_layer_proto == PROTOCOL_UNKNOWN || app_layer_proto == PROTOCOL_POSTGRES) && is_tls(skb, skb_info.data_off, skb_info.data_end, &tls_hdr)) {
```

Comment:
why do we need to modify the original `is_tls` function?
The new function is more complicated and it is not clear we get any added value from the change

If the traffic represents TLS, then apply any further processing needed

Reply:
the new change fills the TLS header struct with some info so we don't have to re-parse it later on

Reply:
I added some extra code to maintain the level of rigidity we had on `main` for the tls handshake validation. I do think it's very useful to set the handshake type in this method so we can use it directly below instead of having to re-do that part